addons/more-links/userscript.js

```diff
@@ -52,6 +52,54 @@ export default async function ({ addon, console }) {
       linkifyTextNode(desc);
       break;
     }
+
+    case "messages":
+      while (true) {
+        const message = await addon.tab.waitForElement(".comment-text p", { markAsSeen: true });
+
+        if (message.textContent.length === 250) {
+          console.log(message);
+          // The message is truncated (unless it happens to be ecxactly 250 characters long), and so links may break. We now fetch the rest of the text for link making
+
+          const username = await addon.auth.fetchUsername();
+
+          const messageInfo = message.parentElement.parentElement.parentElement.firstChild;
+          const link = messageInfo.querySelector("a:not(.social-messages-profile-link)").href;
+
+          async function getContentFromResponse(response, message) {
+            const blob = await response.blob();
+            const usableResponse = JSON.parse(await blob.text());
+
+            // Yes, using innerHTML here, this is so the browser can handle things like &apos; for a comma
+            message.innerHTML = usableResponse.content;
+          }
```

Comment:
`await response.text()` is a thing

also it's not us blocking `.innerHTML`. it's the chrome web store. i imagine they won't allow this cause theoretically what is the API returns some script tag? that's not feasible at all but think of their perspective.

i think there's a function somewhere in the code that sanitizes a string before putting it into innerHTML

Reply:
> `await response.text()` is a thing
> 
> also it's not us blocking `.innerHTML`. it's the chrome web store. i imagine they won't allow this cause theoretically what is the API returns some script tag? that's not feasible at all but think of their perspective.
> 
> i think there's a function somewhere in the code that sanitizes a string before putting it into innerHTML

About the response.text I did that and unless I'm insane, it didn't give me what I needed

I'll look into that function

Reply:
I wouldn't neccessarily expect the CWS to know/believe that though
I believe we once had a release blocked by one of the stores because there we used innerHTML with a constant string, and the string had a commented-out <script> tag.

Reply:
I just had a thought... Could someone not write HTML code in a comment that's longer than 250 characters? This would immediately turn that into real html. What's stopping them from writing a script tag in their comment, the API doesn't need to be compromised at all

Reply:
Oh and also, `await response.json()`?

Reply:
> Oh and also, `await response.json()`?

I swear this is the only way I could do it...I might be going insane, legitimately. Let me check that after my ocd shower

Reply:
> I just had a thought... Could someone not write HTML code in a comment that's longer than 250 characters? This would immediately turn that into real html. What's stopping them from writing a script tag in their comment, the API doesn't need to be compromised at all

Scratch itself also just inserts the HTML without doing anything to it. The API escapes every comment when it's posted.

![Screenshot of a comment saying "<img>"](https://github.com/ScratchAddons/ScratchAddons/assets/130385691/955ac53c-8bc2-4abb-9a76-8d44970d3c79)
![Screenshot of "content": "&lt;img&gt;"](https://github.com/ScratchAddons/ScratchAddons/assets/130385691/f3dcbe3b-4b2a-4329-ab3a-1dd7b5c18ab0)